### ISSUE TITLE
[Feature Request] Add Aria-labels to VTreeview Buttons

### ISSUE THREAD

**Issue by robertzak** [vuetifyjs/vuetify#21585]
### Problem to solve
Currently when using a VTreeview component the Expand/collapse button for each node of the Tree simply read by a Screen Reader as "Button" without any context as to what it does when activated.

### Proposed solution
Add an aria label based on the current state and node label.

So if you have a tree like:

```
> Red
V Green
   > Orange
```
The buttons would read as "Expand Red", "Collapse Green", "Expand Orange"

**Comment by cmooreDT** [comment]
Also please consider adding aria-controls and aria-expanded to the button elements. 

The button to expand Red would would have aria-controls="red-id" and aria-expanded="false", and the v-list-item-title surrounding Red would have id="red-id". When Red is open/expanded, aria-expanded would equal true.
```
<button aria-controls="red-id" aria-expanded="true"> V </button>
<v-list-item-title id="red-id">Red</v-list-item-title>

<button aria-controls="green-id" aria-expanded="false"> > </button>
<v-list-item-title id="green-id">Green</v-list-item-title>
```

**Comment by robertzak** [comment]
I wasn't familiar with those tags (still learning a lot of this stuff as we've been instructed to improve our accessibility). 

@cmooreDT do you know if having those tags make my proposed dynamic labels (IE "Expand Red", "Collapse Green") redundant? So maybe it should just match the list item label instead? Or does the `aria-controls` provide the screen reader enough information that an aria-label isn't needed? Their combo box example had all 3, but I'm not sure if that's relevant for a tree button.


IE should it be:

```
<button aria-controls="red-id" aria-expanded="true" aria-label="Collapse Red"> V </button>
<v-list-item-title id="red-id">Red</v-list-item-title>

<button aria-controls="green-id" aria-expanded="false" aria-label="Expand Green"> > </button>
<v-list-item-title id="green-id">Green</v-list-item-title>
```

Or maybe there should be no aria-labels at all, because a screen reader would then read each node label twice?

**Comment by cmooreDT** [comment]
@robertzak There are 3 component libraries I reference when trying to adhere to WCAG guidelines.

1. [US Web Design Systems](https://designsystem.digital.gov/components/overview/) for US clients specifically
2. [Deque University](https://dequeuniversity.com/library/)
3. [W3C](https://www.w3.org/WAI/ARIA/apg/patterns/)

All 3 of these resources utilize `aria-controls` and `aria-expanded` for accordions. `aria-label` is needed for the screen reader. When button text isn't present, it's what the SR will announce to the user when they are focused on the button. For buttons that are only icons, it is crucial. The exception would be adding an `d-sr-only` span within the icon button. Alternatively, when used on buttons with visible text, it'll override whatever text is already inside the button. This is especially useful if you have a button with something like "Next >" for pagination. The `aria-label` in this instance could be "next page". When focused, the SR would announce "next page" versus "next closed bracket".

In my Vtreeview use case, Red is a link to the Red category page, and > expands the Red category items. The > button would need the `aria-label` to let the SR user know what the button does once it receives focus. In this case, it would be "Expand red" as stated in your initial feature request. As you stated, without the `aria-label`, the SR simply says "button".

`aria-controls` lets any assistive technology know which element will be revealed when the expand button is pressed. For SRs, they would announce something like "panel is expanded" based on `aria-expanded` being true. This would let the user know, information has been revealed and is now present on the screen. The SR will also announce the closing once the use toggles the panel shut. I admit my SR experience is limited as I only use Mac's VoiceOver, but I believe this is standard behavior.

So to go back to your initial questions, `aria-label`, `-controls`, and `-expanded` each offer assistive technologies specific and crucial information. Your [provided example](#issuecomment-3143863514) using all 3 of them is correct. Mine should have included `aria-label` in the V, but I was mainly focused on displaying the relationship between `-controls` and `-expanded`. Hopefully this helped. If not, LMK if you need more clarification. I'm happy to be a resource.

**Comment by robertzak** [comment]
@cmooreDT Thanks for the links/clarification. I've mostly been using W3C as needed to look stuff up. I'll add the other 2 to my bookmarks  and take a look when I'm next working on improving our app's accessibility.
